code/modules/zombie/organs.dm

```diff
@@ -4,90 +4,96 @@
-		qdel(src)
-		return
-	if(owner.suiciding)
-		return
-	if(owner.stat != DEAD && !converts_living)
+	if(!owner)
 		return
+	if(!(src in owner.internal_organs))
+		remove(owner)
 	if(causes_damage && !iszombie(owner) && owner.stat != DEAD)
 		owner.adjustToxLoss(1)
 		if (prob(10))
 			to_chat(owner, "<span class='danger'>You feel sick...</span>")
+	if(timer_id)
+		return
+	if(owner.suiciding)
+		return
+	if(owner.stat != DEAD && !converts_living)
+		return
 	if(!owner.get_int_organ(/obj/item/organ/internal/brain))
 		return
-	if(!iszombie(owner) && !timer_id)
+	if(!iszombie(owner))
 		to_chat(owner, "<span class='cultlarge'>You can feel your heart stopping, but something isn't right... \
 		life has not abandoned your broken form. You can only feel a deep and immutable hunger that \
-		not even death can stop, you will rise again!</span>")	
+		not even death can stop, you will rise again!</span>")
 	var/revive_time = rand(revive_time_min, revive_time_max)
 	var/flags = TIMER_STOPPABLE
 	timer_id = addtimer(CALLBACK(src, .proc/zombify), revive_time, flags)
 
 /obj/item/organ/internal/zombie_infection/proc/zombify()
 	timer_id = null
-	if(owner.stat != DEAD && !converts_living)
+
+	if(!converts_living && owner.stat != DEAD)
 		return
 
 	if(!iszombie(owner))
-		old_species = owner.dna.species
-		dont_delete = TRUE // make sure the remove code above does not run when the species code replaces all the organs
+		old_species = owner.dna.species.type
 		owner.set_species(/datum/species/zombie/infectious)
-		for(var/thing in owner.viruses) // cure any new crit viruses
-			var/datum/disease/D = thing
-			D.cure(0)
-		dont_delete = FALSE
+		for(var/datum/disease/critical/crit in owner.viruses) // cure any new crit viruses
+			crit.cure(0)
+
+	var/stand_up = (owner.stat == DEAD) || (owner.stat == UNCONSCIOUS)
 	//Fully heal the zombie's damage the first time they rise
 	owner.setToxLoss(0)
 	owner.setOxyLoss(0)
 	owner.setBrainLoss(0)
 	owner.setCloneLoss(0)
 	owner.SetLoseBreath(0)
-	owner.heal_overall_damage(INFINITY,INFINITY,TRUE,TRUE,FALSE)
+	owner.heal_overall_damage(INFINITY, INFINITY, TRUE, TRUE, FALSE)
+	owner.setStaminaLoss(0)
 
-	if(owner.update_revive())
-		owner.grab_ghost()
+	if(!owner.update_revive())
+		return
 
-	owner.visible_message("<span class='danger'>[owner] suddenly convulses, as [owner.p_they()] stagger to [owner.p_their()] feet and gain a ravenous hunger in [owner.p_their()] eyes!</span>", "<span class='alien'>You HUNGER!</span>")
-	playsound(owner.loc, 'sound/hallucinations/far_noise.ogg', 50, 1)
+	owner.grab_ghost()
+	owner.visible_message("<span class='danger'>[owner] suddenly convulses, as [owner.p_they()][stand_up ? " stagger to [owner.p_their()] feet and" : ""] gain a ravenous hunger in [owner.p_their()] eyes!</span>", "<span class='alien'>You HUNGER!</span>")
+	playsound(owner.loc, 'sound/hallucinations/far_noise.ogg', 50, TRUE)
 	owner.do_jitter_animation(living_transformation_time)
 	owner.Stun(living_transformation_time)
```

Comment:
stuns for 30 ticks, or a whole minute with the new living_transformation_time variable

Reply:
Yeah, oversight on my part. THanks.